src/pages/index.js

```diff
@@ -73,35 +102,137 @@ function handleImageClick(title, link) {
 }
 
 function makeCard(cardData) {
-  const card = new Card(cardData, cardTemplate, handleImageClick);
+  const card = new Card(
+    cardData,
+    cardTemplate,
+    handleImageClick,
+    handleDeleteClick,
+    handleCardLike
+  );
   return card.generateCard();
 }
 
+function checkIfLiked(ifLiked, element) {
+  if (ifLiked === true) {
+    element
+      .querySelector(".heart-button")
+      .classList.toggle("heart-button_active");
+  }
+}
+
+function handleDeleteClick(cardId) {
+  console.log(cardId);
+  deleteConfirmModal.open(cardId);
+}
+
 // VALIDATION
 const editFormValidator = new FormValidator(
   validationSettings,
   profileEditForm
 );
 const cardFormValidator = new FormValidator(validationSettings, cardAddForm);
+const avatarFormValidator = new FormValidator(
+  validationSettings,
+  avatarEditForm
+);
 
+avatarFormValidator.enableValidation();
 editFormValidator.enableValidation();
 cardFormValidator.enableValidation();
 
 /* EVENT HANDLERS */
 
 function handleProfileEditSubmit({ name, badge }) {
-  user.setUserInfo(name, badge);
-  editProfileModal.close();
+  editProfileModal.setLoading(true, "Saving...");
+  api
+    .updateProfile({ name, badge })
+    .then((message) => {
+      console.log(message);
+      user.setUserInfoOnSubmit(name, badge);
+    })
+    .catch((err) => {
+      console.err(err);
+    })
+    .finally(() => {
+      editProfileModal.setLoading(false, "Save");
+      editProfileModal.close();
```

Comment:
`      editProfileModal.close();
`

You need to close modals only in blocks `then` after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server)